src/lib/Hydra/Helper/AddBuilds.pm

```diff
@@ -76,7 +76,12 @@ sub handleDeclarativeJobsetBuild {
             push @kept, ".jobsets";
             $project->jobsets->search({ name => { "not in" => \@kept } })->update({ enabled => 0, hidden => 1 });
             while ((my $jobsetName, my $spec) = each %$declSpec) {
-                updateDeclarativeJobset($db, $project, $jobsetName, $spec);
+                eval {
+                    updateDeclarativeJobset($db, $project, $jobsetName, $spec);
+                };
+                if ($@) {
+                    print STDERR "ERROR: failed to process declarative jobset ", $project->name, ":${jobsetName}, ", $@, "\n";
+                }
```

Comment:
The old behavior caused the exception from updateDeclarativeJobset to percolate up, and this one prints a warning and swallows it. Is this an intended behavior change?

Reply:
If we wanted to preserve the behavior (I sort of suspect, cautiously, that we do), we'd add:

```
die $@
```

Reply:
there was a declarative jobset, that has invalid parameters on it, and because of the old behavior, one jobset being invalid stopped the entire project from updating

this new behavior makes it capable of updating other jobsets when some are invalid, so the entire project doesnt grind to a total halt